cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -601,15 +547,17 @@ struct ConvertFunctor {
                                                       parse_options_view const& opts,
                                                       bool as_hex = false)
   {
-    static_cast<T*>(out_buffer)[row] = [as_hex, &opts, begin, end]() -> T {
+    auto const value = [as_hex, &opts, begin, end]() -> std::optional<T> {
       // Check for user-specified true/false values
       auto const field_len = static_cast<size_t>(end - begin);
       if (serialized_trie_contains(opts.trie_true, {begin, field_len})) { return 1; }
       if (serialized_trie_contains(opts.trie_false, {begin, field_len})) { return 0; }
-      return as_hex ? decode_value<T, 16>(begin, end, opts) : decode_value<T>(begin, end, opts);
+      return as_hex ? cudf::io::parse_numeric<T, 16>(begin, end, opts)
```

Comment:
Correct me if I'm wrong, it seems like in dropping `decode_value` we lost support for duration and timestamp types. Is that intentional, is it OK because those weren't actually supported before and those code paths were never taken, or is it an intentional breaking change?

Reply:
Ah it looks like the branches where we could hit duration or timestamp types have had the logic inlined below, so maybe this is fine then.